src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/RejectedEnvelopesReportServiceTest.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services.reports;
+
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.ListBlobItem;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.reports.models.RejectedEnvelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor.BlobManager;
+
+import java.net.URI;
+import java.util.EnumSet;
+import java.util.List;
+
+import static com.microsoft.azure.storage.blob.BlobListingDetails.SNAPSHOTS;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RejectedEnvelopesReportServiceTest {
```

Comment:
can we please add a test case when there are no rejected envelopes ?

Reply:
done